torch/csrc/jit/script/sugared_value.h

```diff
@@ -141,6 +142,10 @@ struct TORCH_API SimpleValue : public SugaredValue {
     return value_;
   }
 
+  virtual std::shared_ptr<SimpleValue> new_with(Value* new_value) {
```

Comment:
in person: let's just make ModuleValue::setAttr do `SimpleValue(self_).setAttr` that would avoid these issues of needing to propagate SimpleValue subclasses around.